### PR TITLE
Add Traversable support of length filter

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/twigphp/Twig/issues/2922

### CODE REVIEW

File: src/Extension/CoreExtension.php [hunk 1]
```diff
@@ -1298,22 +1298,18 @@ function twig_length_filter(Environment $env, $thing)
         }
 
         if (is_scalar($thing)) {
-            return mb_strlen($thing, $env->getCharset());
+            return mb_strlen((string) $thing, $env->getCharset());
```
Comment (fabpot) [inline src/Extension/CoreExtension.php]: I think this can be reverted.

File: src/Extension/CoreExtension.php [hunk 2]
```diff
@@ -1298,22 +1298,18 @@ function twig_length_filter(Environment $env, $thing)
...
+        if (\is_object($thing) && method_exists($thing, '__toString')) {
             return mb_strlen((string) $thing, $env->getCharset());
         }
 
-        if ($thing instanceof \Countable || \is_array($thing)) {
-            return \count($thing);
-        }
-
-        if ($thing instanceof \IteratorAggregate) {
+        if ($thing instanceof \Traversable) {
```
Comment (fabpot) [inline src/Extension/CoreExtension.php:1308]: I would do that before using `__toString` (also to be BC).

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Extension/CoreExtension.php (hunk 2 above)
Comment (SpacePossum) [reply src/Extension/CoreExtension.php:1308]: good catch, done and I've added update the test to catch this in the future : )

File: src/Extension/CoreExtension.php (hunk 1 above)
Comment (SpacePossum) [reply src/Extension/CoreExtension.php]: :+1:  done

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @SpacePossum.

**Pull request merged**